torch/csrc/utils/tensor_new.cpp

```diff
@@ -211,10 +212,15 @@ Tensor internal_new_from_data(const Type & type, at::optional<Device> device_opt
                                                        *torch::getLayout(type.backend()),
                                                        type_inference_device_type);
       const auto& type_to_use = type_inference ? type_inference_type : type;
-      return copy_variables ? new_with_tensor_copy(type_to_use, var, device_index) :
+      // temporarily mute requires_grad when copy from var with requires_grad = false
```

Comment:
Eeeeew. I don't like it. What if a concurrent thread is reading from this variable? Surely there's a way to do this without mutating.